### PR TITLE
[ROR 1.25.0][RORDEV-365] auth_key_pbkdf2 description

### PR DESCRIPTION
**Pull request by coutoPL**
[don't merge before release of ROR 1.25.0]

ref: https://github.com/sscarduzio/elasticsearch-readonlyrest-plugin/pull/651

### CODE REVIEW

**Comment by coutoPL** [comment]
ref: https://github.com/sscarduzio/elasticsearch-readonlyrest-plugin/pull/651

File: elasticsearch.md [hunk 1]
```diff
@@ -1124,13 +1124,45 @@ Local ReadonlyREST users are authenticated via HTTP Basic Auth. This authenticat
...
+Accepts [HTTP Basic Auth](https://en.wikipedia.org/wiki/Basic_access_authentication). The value is a string like `username:password` _hashed in_ [_SHA512_](https://md5calc.com/hash/sha512). Clients will need to provide the usual Authorization header. 
+
+There are also available other rules with less secure SHA algorithms `auth_key_sha256` and `auth_key_sha1`.
+
+The rules support also alternative syntax, where only password is hashed, eg:
+
+`auth_key_sha512: "admin:280ac6f...94bf9"`
+
+In the example below `admin` is a username and `280ac6f...94bf9` is
+hashed secret.
```
Comment (sscarduzio) [inline elasticsearch.md:1142]: in the example above, `admin` is the username, and `...` is the hashed password.

File: elasticsearch.md [hunk 2]
```diff
@@ -1124,13 +1124,45 @@ Local ReadonlyREST users are authenticated via HTTP Basic Auth. This authenticat
...
+
+`auth_key_pbkdf2: "KhIxF5EEYkH5GPX51zTRIR4cHqhpRVALSmTaWE18mZEL2KqCkRMeMU4GR848mGq4SDtNvsybtJ/sZBuX6oFaSg=="` # logstash:logstash 
+
+`auth_key_pbkdf2: "logstash:JltDNAoXNtc7MIBs2FYlW0o1f815ucj+bel3drdAk2yOufg2PNfQ51qr0EQ6RSkojw/DzrDLFDeXONumzwKjOA=="` # logstash:logstash
+
+Accepts [HTTP Basic Auth](https://en.wikipedia.org/wiki/Basic_access_authentication). 
+The value is hashed in the same way as it's done in
+`auth_key_sha512` rule, but it uses [_PBKDF2_](https://en.wikipedia.org/wiki/PBKDF2) 
+key derivation function. At the moment there is no way to configure
+it, so during hash generation user has to take into consideration
```
Comment (sscarduzio) [inline elasticsearch.md:1154]: during the hash generation process, the user has to take in consideration the following ...

File: elasticsearch.md [hunk 3]
```diff
@@ -1124,13 +1124,45 @@ Local ReadonlyREST users are authenticated via HTTP Basic Auth. This authenticat
...
+
+| Input parameter | Value | Comment |
+| --------------- | ----- | ------- |
+| Pseudorandom function | HmacSHA512 | |
+| Salt | use hashed value as a salt | eg. hashed value = `logstash:logstash`, use `logstash:logstash` as the salt |
+| Iterations count | 10000 | |
+| Derived key length | 512 | bits |
+<br>
+
+The hash can be calculated using [this calculator](https://8gwifi.org/pbkdf.jsp) (notice that salt have to base Base64 encoded).
```
Comment (sscarduzio) [inline elasticsearch.md:1165]: notice that the salt has to be Base64 encoded

**Review by sscarduzio — CHANGES_REQUESTED** [review]
Just a few small corrections.

**Pull request merged**